dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -552,39 +529,52 @@ void FramesDecoder::SeekFrame(int frame_id) {
   //  - for CFR, when we know pts, but don't know keyframes
 
   DALI_ENFORCE(
-    frame_id >= 0 && frame_id < NumFrames(),
-    make_string("Invalid seek frame id. frame_id = ", frame_id, ", num_frames = ", NumFrames()));
+      frame_id >= 0 && frame_id < NumFrames(),
+      make_string("Invalid seek frame id. frame_id = ", frame_id, ", num_frames = ", NumFrames()));
 
-  auto &frame_entry = Index(frame_id);
-  int keyframe_id = frame_entry.last_keyframe_id;
-  auto &keyframe_entry = Index(keyframe_id);
+  if (frame_id == next_frame_idx_) {
+    return;  // No need to seek
+  }
 
-  LOG_LINE << "Seeking to frame " << frame_id << " timestamp " << frame_entry.pts << std::endl;
+  // If we have an index, we can seek to the nearest keyframe first
+  if (HasIndex()) {
+    const auto &current_frame = Index(next_frame_idx_);
+    const auto &requested_frame = Index(frame_id);
+    auto keyframe_id = requested_frame.last_keyframe_id;
+    if (current_frame.last_keyframe_id != keyframe_id) {
```

Comment:
I think it should be:
```
if (requested_frame.last_keyframe_id > next_frame_idx_)
```
So it guarantees we skip frames when seeking.

Reply:
we would still want to use seeking to jump to a far ahead frame, wouldn't we?

Reply:
Yes, but we need to make sure that the key frame corresponding to this `far ahead frame` is ahead of the current frame, and the condition I proposed should guarantee that. Otherwise we may end up rewinding earlier than the current frame is.

Reply:
we may still want to seek back, if the requested frame is before the current one. Am I missing something?

Reply:
So how about:
```
if (requested_frame.last_keyframe_id > next_frame_idx_ || frame_id < next_frame_idx_ )
```